watertap/property_models/coagulation_prop_pack.py

```diff
@@ -0,0 +1,748 @@
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+"""
+Initial property package for water treatment via a Coagulation-Flocculation process
+"""
+from pyomo.environ import (Constraint,
+                           Var,
+                           Param,
+                           Expression,
+                           Reals,
+                           NonNegativeReals,
+                           Suffix,
+                           value,
+                           assert_optimal_termination,
+                           check_optimal_termination)
+from pyomo.environ import units as pyunits
+
+# Import IDAES cores
+from idaes.core import (declare_process_block_class,
+                        MaterialFlowBasis,
+                        PhysicalParameterBlock,
+                        StateBlockData,
+                        StateBlock,
+                        MaterialBalanceType,
+                        EnergyBalanceType)
+from idaes.core.components import Component
+from idaes.core.phases import LiquidPhase, SolidPhase, PhaseType
+from idaes.core.util.initialization import fix_state_vars, revert_state_vars, solve_indexed_blocks
+from idaes.core.util.model_statistics import degrees_of_freedom, number_unfixed_variables
+from idaes.core.util.exceptions import PropertyPackageError, InitializationError, ConfigurationError
+import idaes.core.util.scaling as iscale
+import idaes.logger as idaeslog
+from idaes.core.util import get_solver
+
+__author__ = "Austin Ladshaw"
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+# Forward declaration of 'CoagulationParameterData'
+@declare_process_block_class("CoagulationParameterBlock")
+class CoagulationParameterData(PhysicalParameterBlock):
+    CONFIG = PhysicalParameterBlock.CONFIG()
+
+    def build(self):
+        """
+        Callable method for Block construction.
+        """
+        super(CoagulationParameterData, self).build()
+
+        self._state_block_class = CoagulationStateBlock
+
+        # phases
+        self.Liq = LiquidPhase()
+        self.Sol = SolidPhase()
+
+        # components
+        self.H2O = Component(default={"valid_phase_types": PhaseType.liquidPhase})
+        self.TSS = Component(default={"valid_phase_types": PhaseType.liquidPhase})
+        self.TDS = Component(default={"valid_phase_types": PhaseType.liquidPhase})
+        self.Sludge = Component(default={"valid_phase_types": PhaseType.solidPhase})
```

Comment:
I vaguely remember us talking about this at a meeting but don't recall the consensus. From my perspective, I don't see why we _need_ sludge as a separate solid phase component.

Reply:
Yes, I feel we need to have separate phases. This is thinking forward to the settling tank unit model where we settle out only the solids. In a general sense, the solids won't all necessarily be called 'Sludge', so it will be easier to settling out materials by just identifying all components in the solid phase as the things that get settled. 

For instance, if we want to use the settling tank unit with a Generic Property package that is calculating precipitates. There are many precipitates possible, but they will always be identified as solid phase. So we could using a generalized approach in the settling tank that would work with any solids just by looking at what components are identified as solid based on the phase it is declared in.

Reply:
So, in my mind, that only makes sense if we assume all the solids are removed (i.e., for a perfect separator). In the 0D settling tank, we may have only a specific amount of solids settled, the rest may remain in the stream.